WWTExplorer3d/ReferenceFrame.cs

```diff
@@ -440,27 +464,29 @@ public virtual void InitializeFromXml(XmlNode node)
 
             if (ReferenceFrameType == ReferenceFrameTypes.Orbital)
             {
-                SemiMajorAxis = Double.Parse(node.Attributes["SemiMajorAxis"].Value);
-                SemiMajorAxisUnits = (AltUnits)Enum.Parse(typeof(AltUnits), node.Attributes["SemiMajorAxisScale"].Value);
-
                 Eccentricity = Double.Parse(node.Attributes["Eccentricity"].Value);
                 Inclination = Double.Parse(node.Attributes["Inclination"].Value);
                 ArgumentOfPeriapsis = Double.Parse(node.Attributes["ArgumentOfPeriapsis"].Value);
                 LongitudeOfAscendingNode = Double.Parse(node.Attributes["LongitudeOfAscendingNode"].Value);
                 MeanAnomolyAtEpoch = Double.Parse(node.Attributes["MeanAnomolyAtEpoch"].Value);
                 MeanDailyMotion = Double.Parse(node.Attributes["MeanDailyMotion"].Value);
                 Epoch = Double.Parse(node.Attributes["Epoch"].Value);
-
+                SemiMajorAxis = Double.Parse(node.Attributes["SemiMajorAxis"].Value);
+                SemiMajorAxisUnits = (AltUnits)Enum.Parse(typeof(AltUnits), node.Attributes["SemiMajorAxisScale"].Value);
             }
             if (ReferenceFrameType == ReferenceFrameTypes.Trajectory)
             {
                 if (node["Trajectory"] != null)
                 {
                     foreach (XmlNode child in node["Trajectory"].ChildNodes)
                     {
-                        Trajectory.Add(new TrajectorySample(child.InnerText));
+                        Trajectory.Add(new TrajectorySample(child.InnerText, SemiMajorAxisUnits));
                     }
                 }
+                if (node.Attributes["TrajectoryUnits"] != null)
+                {
+                    TrajectoryUnits = (AltUnits)Enum.Parse(typeof(AltUnits), node.Attributes["TrajectoryUnits"].Value);
+                }
```

Comment:
What's the default value of `SemiMajorAxisUnits` here? I think this is doing the right thing regarding compatibility with older versions of the software, but want to make sure:

- If we read a file created by an older version of the software, will we get the right results?
- If an older version of the software reads a file created by us, will we get the right results to the greatest extent possible?

Reply:
I think the second bullet point is a problem, as this is right now. We'll need to serialize everything in meters in order for older versions to correctly read new files, since they won't know to look for the TrajectoryUnits field.